### PR TITLE
Implement feedback on Mint Access Control

### LINKED ISSUE #5: Update MinterAccessControl to use Enumerable AddressSet
The current implementation with a `mapping(address => bool)` can't be used to enumerate the members of the minter roles.
`EnumerableSetUpgradeable.AddressSet` will be more flexible and can be reused to implement #3

### PR DESCRIPTION
**Pull request by Hadrienlc**
related to https://github.com/liteflow-labs/rarible-protocol-contracts/issues/2

### CODE REVIEW

File: tokens/contracts/erc-1155/ERC1155Lazy.sol [hunk 1]
```diff
@@ -10,7 +10,7 @@ import "@rarible/lazy-mint/contracts/erc-1155/IERC1155LazyMint.sol";
 import "./Mint1155Validator.sol";
 import "./ERC1155BaseURI.sol";
 
-abstract contract ERC1155Lazy is IERC1155LazyMint, ERC1155BaseURI, Mint1155Validator, RoyaltiesV2Upgradeable, RoyaltiesV2Impl {
+abstract contract ERC1155Lazy is IERC1155LazyMint, ERC1155BaseURI, Mint1155Validator, RoyaltiesV2Upgradeable, RoyaltiesV2Impl  {
```
Comment (NicolasMahe) [inline tokens/contracts/erc-1155/ERC1155Lazy.sol:13]:
```suggestion
abstract contract ERC1155Lazy is IERC1155LazyMint, ERC1155BaseURI, Mint1155Validator, RoyaltiesV2Upgradeable, RoyaltiesV2Impl {
```

File: tokens/contracts/erc-1155/ERC1155Rarible.sol [hunk 2]
```diff
@@ -49,7 +51,7 @@ contract ERC1155Rarible is ERC1155Base {
 
     function mintAndTransfer(LibERC1155LazyMint.Mint1155Data memory data, address to, uint256 _amount) public override {
         if (isPrivate){
-          require(owner() == data.creators[0].account, "minter is not the owner");
+            require(owner() == data.creators[0].account || isValidMinter(data.creators[0].account), "minter not granted or not owner");
```
Comment (NicolasMahe) [inline tokens/contracts/erc-1155/ERC1155Rarible.sol:54]:
```suggestion
          require(owner() == data.creators[0].account || isValidMinter(data.creators[0].account), "minter not granted or not owner");
```

File: tokens/test-access-control.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+truffle test ./test/erc-721-minimal/RaribleUser.test.js \
+             ./test/erc-1155/ERC1155RaribleUser.test.js \
+             ./test/MinterAccessControl.test.js \
+             ./test/contracts/access/MinterAccessControlTest.sol
```
Comment (NicolasMahe) [inline tokens/test-access-control.sh:5]: can you remove this script if it is only useful for your local test?

File: tokens/contracts/access/MinterAccessControl.sol [hunk 4]
```diff
@@ -10,40 +10,18 @@ contract MinterAccessControl is Initializable, OwnableUpgradeable {
...
-    bool public minterAccessControlEnabled;
-
-    event MinterAccessControlEnabled();
-    event MinterAccessControlDisabled();
+    
     event MinterGranted(address indexed account);
     event MinterRevoked(address indexed account);
 
-    function __MinterAccessControl_init() internal initializer {
+    function __MinterAccessControl_init() external initializer {
```
Comment (antho1404) [inline tokens/contracts/access/MinterAccessControl.sol:17]: I'm not sure what's this `external` is for. It feels like this should not be something that we want to expose, is it?
I see this is used in tests but should it be done for production?

File: tokens/contracts/erc-1155/ERC1155Lazy.sol [hunk 5]
```diff
@@ -60,7 +59,6 @@ abstract contract ERC1155Lazy is IERC1155LazyMint, ERC1155BaseURI, Mint1155Valid
         address sender = _msgSender();
 
         require(minter == sender || isApprovedForAll(minter, sender), "ERC1155: transfer caller is not approved");
-        require(isValidMinter(minter), "ERC1155: minter not granted");
```
Comment (antho1404) [inline tokens/contracts/erc-1155/ERC1155Lazy.sol:63]: Just to confirm, is this ok to remove that as the function is still public?
It's being overridden by the parent contract, but I just want to make sure that's ok.

**Review by antho1404 — APPROVED** [review]

File: tokens/contracts/erc-1155/ERC1155Lazy.sol (hunk 5 above)
Comment (Hadrienlc) [reply tokens/contracts/erc-1155/ERC1155Lazy.sol:63]: `isValidMinter()` is a call we added and points to `MinterAccessControl` which is not available anymore in this class.

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**